lib/drivers/bayerContourNext.js

```diff
@@ -0,0 +1,510 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2014, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+/**
+ * IOET's code start here
+ * packet frame format <STX> FN text <ETX> C1 C2 <CR> <LF>
+ * */
+var _ = require('lodash');
+var async = require('async');
+var sundial = require('sundial');
+var crcCalculator = require('../crc.js');
+var struct = require('../struct.js')();
+var annotate = require('../eventAnnotations');
+
+var isBrowser = typeof window !== 'undefined';
+var debug = isBrowser ? require('../bows')('BCNextDriver') : debug;
+
+
+module.exports = function (config) {
+  var cfg = _.clone(config);
+  var hidDevice = config.deviceComms;
+
+  var STX = 0x02;
+  var ETB = 0x17;
+  var ETX = 0x03;
+  var EOT = 0x04;
+  var HID_PACKET_SIZE = 60;
+
+  /*
+   * TODO This info should be collected in the initial phase cause we can't assert if
+   *      it is the same for all contour devices
+   * */
+  var REPORT_BYTES = {
+    reportID: 0x41, // A
+    checksum: 0x00,
+    hostID  : 0x42, // B
+    deviceID: 0x43  // C
+  };
+
+  /* end */
+
+  var ASCII_CONTROL = {
+    ACK : 0x06,
+    NAK : 0x15,
+    ENQ : 0x05
+  };
+
+  var probe = function(cb){
+    debug('attempting probe of Bayer Contour Next');
+  };
+
+  var bcnPacketHandler = function (buffer) {
+    // we need to get the REPORT_BYTES here for the first communication
+    // time if they are not set
+    var discardCount = 3;
```

Comment:
I don't see how this can be correct. If you want to always discard the first 3 bytes of the packet, why the conditional? And why not just set the minimum packet size to 9 and handle those 3 bytes in the packet processor? At the very least this doesn't make sense and it seems like it must be wrong.

If it's NOT wrong, then you need to put a comment explaining why you're doing something so strange.